archive/common/src/test/scala/uk/ac/wellcome/platform/archive/common/progress/fixtures/ProgressGenerators.scala

```diff
@@ -61,6 +68,20 @@ trait ProgressGenerators extends RandomThings {
     ProgressStatusUpdate(id, status, maybeBag, events)
   }
 
+  def createProgressBagUpdateWith(id: UUID,
+                                  bagId: BagId,
+                                  status: Status = Progress.Processing,
+                                  events: Seq[ProgressEvent] = List(
+                                    createProgressEvent)): ProgressUpdate = {
+    ProgressStatusUpdate(id, status, Some(bagId), events)
+  }
+
+  def createBagId = BagId(createStorageSpace, createExternalIdentifier)
+
+  def createStorageSpace = StorageSpace(randomAlphanumeric())
+
+  def createExternalIdentifier = ExternalIdentifier(randomAlphanumeric())
```

Comment:
This already exists – see `ExternalIdentifierGenerators`.

Reply:
thanks